### PR TITLE
Convert to new style mkdocs

### LINKED ISSUE #88: Readme for adding a parser.
### Environment
* netutils version:  

A readme file in configs that provides the proper understanding of how to write a new parser would be helpful.

### Use Case
External contributor wants to add support for a new parser, it'd be nice for a getting started how to guide.

### PR DESCRIPTION
**Pull request by itdependsnetworks**
Fixes  #88, #127, #131, and #150

### CODE REVIEW

**Comment by jeffkala** [comment]
should `code_references` be nested under `/docs/dev` or just `/docs`?

**Comment by itdependsnetworks** [comment]
> should `code_references` be nested under `/docs/dev` or just `/docs`?

This is the standard already, I just followed it.

File: README.md [hunk 1]
```diff
@@ -1,235 +1,40 @@
...
+- [Administrator Guide](https://netutils.readthedocs.io/en/latest/admin/install/) - How to Install, Configure, Upgrade, or Uninstall the library.
+- [Developer Guide](https://netutils.readthedocs.io/en/latest/dev/contributing/) - Extending the library, Code Reference, Contribution Guide.
+- [Release Notes / Changelog](https://netutils.readthedocs.io/en/latest/admin/release_notes/).
+- [Frequently Asked Questions](https://netutils.readthedocs.io/en/latest/user/faq/).
 
-# Installation
+### Contributing to the Docs
 
-Option 1: Install from PyPI.
+You can find all the Markdown source for the library documentation under the [docs](https://github.com/networktocode/netutils/tree/develop/docs) folder in this repository. For simple edits, a Markdown capable editor is sufficient - clone the repository and edit away.
```
Comment (jeffkala) [inline README.md:32]:
```suggestion
All the Markdown source for the library documentation can be found under the [docs](https://github.com/networktocode/netutils/tree/develop/docs) folder in this repository. For simple edits, a Markdown capable editor is sufficient - clone the repository and edit away.
```

File: docs/dev/dev_config.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+There are a series of considerations documented below, when developing a new parser.
+
+- Creation of a new class that must be created in `netutils/config/parser.py` file.
+- Creation of a parser class that inherits from the class `BaseConfigParser` in the Python Method Resolution Order (MRO).
+    - In nearly all cases should inherit directory off of `BaseSpaceConfigParser` or `BaseBraceConfigParser`.
+    - `BaseSpaceConfigParser` is for Cisco IOS-like configurations.
+    - `BaseBraceConfigParser` is for JUNOS-like configurations that use curly braces.
+- Create the class name in the format of `{os_name.title()}ConfigParser`.
+    - The class's `__init__` method must keep true to the signature or `__init__(self, config)`.
```
Comment (jeffkala) [inline docs/dev/dev_config.md:62]:
```suggestion
    - The classes `__init__` method must keep true to the signature or `__init__(self, config)`.
```

File: docs/dev/dev_environment.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Building Your Development Environment
+
+## Quickstart
+
+The development environment can be used in two ways:
+
+1. `Recommended` All services are spun up using Docker and a local mount so you can develop locally, but Nautobot is spun up within the Docker container.
```
Comment (jeffkala) [inline docs/dev/dev_environment.md:7]:
```suggestion
1. `Recommended` All services are spun up using Docker and a local mount so you can develop locally, but Netutils is spun up within the Docker container.
```

File: docs/dev/dev_environment.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+```shell
+poetry shell
+poetry install
+cp development/creds.example.env development/creds.env
+invoke build
+invoke start
+```
+
+The Nautobot server can now be accessed at [http://localhost:8080](http://localhost:8080) and the live documentation at [http://localhost:8001](http://localhost:8001).
```
Comment (jeffkala) [inline docs/dev/dev_environment.md:40]:
```suggestion
Live documentation can be viewed at [http://localhost:8001](http://localhost:8001).
```

File: docs/dev/dev_environment.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+To either stop or destroy the development environment use the following options.
+
+- **invoke stop** - Stop the containers, but keep all underlying systems intact
+- **invoke destroy** - Stop and remove all containers, volumes, etc. (This results in data loss due to the volume being deleted)
+
+## Poetry
+
+Poetry is used in lieu of the "virtualenv" commands and is leveraged in both environments. The virtual environment will provide all of the Python packages required to manage the development environment such as **Invoke**. See the [Local Development Environment](#local-poetry-development-environment) section to see how to install Nautobot if you're going to be developing locally (i.e. not using the Docker container).
```
Comment (jeffkala) [inline docs/dev/dev_environment.md:49]:
```suggestion
Poetry is used in lieu of the "virtualenv" commands and is leveraged in both environments. The virtual environment will provide all of the Python packages required to manage the development environment such as **Invoke**. See the [Local Development Environment](#local-poetry-development-environment) section to see how to install Netutils if you're going to be developing locally (i.e. not using the Docker container).
```

**Review by jeffkala — CHANGES_REQUESTED** [review]

**Comment by qduk** [comment]
I may just be missing it, but I don't see the functions for each file. Something equivalent to [this](https://netutils.readthedocs.io/en/latest/netutils/bandwidth/index.html)

**Comment by itdependsnetworks** [comment]
> I may just be missing it, but I don't see the functions for each file. Something equivalent to [this](https://netutils.readthedocs.io/en/latest/netutils/bandwidth/index.html)

https://github.com/networktocode/netutils/pull/152/files#diff-34ec211d37292b70f8207ed7f4785ca06077f5ae7d67f5e8549ec5b4e47f20a0R5

**Review by qduk — APPROVED** [review]
LGTM

**Pull request merged**